### ISSUE TITLE
YOLOv11 to Pytorch

### ISSUE THREAD

**Issue by obinopaul** [ultralytics/ultralytics#17578]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have been working with YOLO for some time, primarily using the Ultralytics library. Recently, I started conducting research and aim to extract the model architecture and weights to use with PyTorch. Could you assist me with this?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @obinopaul, thank you for your interest in Ultralytics 🚀! We recommend checking out our [Docs](https://docs.ultralytics.com) for new users to explore [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples, which might provide the guidance you need.

If you're working on extracting model architectures and weights for PyTorch, this is a fantastic opportunity to dive deep into the Ultralytics library! While I'm here to guide you, an Ultralytics engineer will also hop in soon to provide more detailed assistance.

If this is a 🐛 Bug Report, ensure you provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to assist us with debugging.

For your custom training ❓question, please supply as much context as possible, such as dataset images and training logs, and make sure you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community with real-time chat on [Discord](https://ultralytics.com/discord) 🎧. For deeper discussions, visit [Discourse](https://community.ultralytics.com), or check out our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Make sure you upgrade to the latest `ultralytics` package, which includes all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can run in verified environments with dependencies like [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing, ensuring proper function of YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu.

**Comment by Y-T-G** [comment]
`ultralytics` uses PyTorch. PyTorch is simply a framework. So when you say, you want to extract them to PyTorch, I guess you want to get one YOLO huge class with module definitions. But the modules used in YOLO are custom. They don't come with PyTorch, but are built using PyTorch. So you can't load it in PyTorch without redefining all of those modules first.

If you just want to infer with PyTorch without `ultralytics`, you can export to `torchscript`>

https://docs.ultralytics.com/modes/export/

**Comment by MegaGimen** [comment]
So how to expert?

**Comment by glenn-jocher** [comment]
@MegaGimen You can export your YOLO11 model to TorchScript format using:

```python
from ultralytics import YOLO
model = YOLO("yolo11n.pt")
model.export(format="torchscript")
```

This creates a `.torchscript` file that you can load and use with PyTorch without the ultralytics library.
